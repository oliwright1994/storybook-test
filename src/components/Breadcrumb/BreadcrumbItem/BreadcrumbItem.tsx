import { Space } from '@bright/constants'
import { BreadcrumbItem as ChakraBreadcrumbItem, BreadcrumbLink as ChakraBreadcrumbLink } from '@chakra-ui/core'
import React, { FC } from 'react'

interface IBreadcrumbItem {
  isCurrentPage?: boolean
  href: string
}

const Breadcrumb: FC<IBreadcrumbItem> = ({ children, isCurrentPage = false, href }) => {
  return (
    <ChakraBreadcrumbItem addSeparator={!isCurrentPage} spacing={Space.s} separator="/" isCurrentPage={isCurrentPage}>
      <ChakraBreadcrumbLink href={href}>{children}</ChakraBreadcrumbLink>
    </ChakraBreadcrumbItem>
  )
}

export default Breadcrumb
