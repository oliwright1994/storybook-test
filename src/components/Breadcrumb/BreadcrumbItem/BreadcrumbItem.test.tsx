import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import BreadcrumbItem from '.'

describe('<BreadcrumbItem />', () => {
  const defaultProps = {
    href: '/home',
    children: 'Link Text',
  }

  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(<BreadcrumbItem {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  it('should render an <a> tag with the correct href', () => {
    const { getByText } = renderWithBrightTheme(<BreadcrumbItem {...defaultProps} />)
    const link = getByText(defaultProps.children)

    expect(link.nodeName).toEqual('A')
    expect(link).toHaveAttribute('href', '/home')
  })

  it('should add aria-current attribute when isCurrentPage is true', () => {
    const { getByText } = renderWithBrightTheme(<BreadcrumbItem isCurrentPage {...defaultProps} />)
    const link = getByText(defaultProps.children)

    expect(link).toHaveAttribute('aria-current', 'page')
  })
})
