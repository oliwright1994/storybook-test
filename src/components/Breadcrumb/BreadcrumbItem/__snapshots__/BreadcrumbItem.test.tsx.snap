// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadcrumbItem /> should render as expected 1`] = `
<div>
  <li
    class="css-18biwo"
  >
    <a
      class="css-u5zpo1"
      href="/home"
    >
      Link Text
    </a>
    <span
      class="css-1ho43uv"
      role="presentation"
    >
      /
    </span>
  </li>
</div>
`;
